### PR TITLE
Migrate to OpenAPI 3.1

### PR DESCRIPTION
**Pull request by ai-oleynikov**
This PR updates OpenAPI version from 3.0 to 3.1 in accordance with this guide: https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0

The only changes I made are:
- deletion of `"nullable": false` as this keyword is now deprecated, and nulls are rejected by default anyway
- replacement of `"type": "<some-type>", "nullable": true` with `"type": ["<some-type>", "null"]`
- replacement of `"minimum": 0, "exclusiveMinimum": true` with `"exclusiveMinimum": 0`
- replacement of `"example"` with `"examples"` keyword inside `Schema Objects`

### CODE REVIEW

File: BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json [hunk 1]
```diff
@@ -176,11 +176,11 @@
...
                 ],
                 "operationId": "Apps_GetCrowdfundApp",
-                "summary": "Get crowdfund app data",
-                "description": "Returns crowdfund app data",
+                "summary": "Get crowdfunding app data",
+                "description": "Returns crowdfunding app data",
                 "responses": {
                     "200": {
-                        "description": "Crowdfund app data",
+                        "description": "Crowdfunding app data",
```
Comment (ai-oleynikov) [inline BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json:183]: This one change is unrelated to the migration to OpenAPI 3.1. I just couldn't ignore it. "Crowdfunding app" sounds more natural. Feel free to revert this change if you disagree though.

**Review by dennisreimann — APPROVED** [review]
LGTM

**Comment by Kukks** [comment]
The swagger schema test fails. See CheckSwaggerIsConformToSchema

**Review by Kukks — CHANGES_REQUESTED** [review]
The swagger schema test fails. Please check CheckSwaggerIsConformToSchema

**Comment by dennisreimann** [comment]
It looks like the Newtonsoft `JSchema` package has problems parsing th OpenAPI 3.1 spec. Updated the lib and tried [various approaches](https://www.newtonsoft.com/jsonschema/help/html/LoadingSchemas.htm), but it fails with `Error when resolving schema reference '#/$defs/info'. Path 'properties.info', line 11, position 13`, which one can also replicate with their [JSON Schema Validator](https://www.jsonschemavalidator.net/) (select the `openapi.json` from the dropdown).

**Comment by dennisreimann** [comment]
Rebased and tried to make the schema validation work by removing the [`$id` attribute](https://json-schema.org/understanding-json-schema/structuring#dollarid), which prevented that the references worked in general. But now there is still an error with nested `$defs`: `Could not resolve schema reference '#/$defs/parameter/dependentSchemas/schema/$defs/styles-for-path'. Path '$defs.parameter.dependentSchemas.schema.allOf[1]'`

@ai-oleynikov Any idea how we can resolve this?

**Comment by dennisreimann** [comment]
@ai-oleynikov Any idea how we can resolve this?

**Comment by pavlenex** [comment]
ping @ai-oleynikov still interested in helping with this one, if so can you check ^

**Comment by ai-oleynikov** [comment]
Sorry for my disappearance! I'll work on fixing this today and report back if I have any issues I can't resolve.

**Comment by ai-oleynikov** [comment]
The problem seems to be that JSchema doesn't support the current version of the JSON Schema spec.

https://www.jsonschemavalidator.net/ says the following

> An online, interactive JSON Schema validator. Supports JSON Schema Draft 3, Draft 4, Draft 6, Draft 7 and Draft 2019-09.

Therefore. it can't parse JSON Schema 2020-12. The error messages @dennisreimann shared above indicate that JSchema is unable to resolve references containing $defs, likely due to the [changes](https://json-schema.org/draft/2020-12/release-notes#embedded-schemas-and-bundling) in the behavior of $defs introduced in JSON Schema 2020-12.

Unfortunately, the OpenAPI Initiative doesn't provide a JSON Schema for OpenAPI 3.1 using spec versions other than 2020-12. All their [schema files](https://github.com/OAI/OpenAPI-Specification/tree/gh-pages/oas/3.1/schema) are written using this version.

I'm not sure what can be done here, except replacing JSchema with another JSON Schema validator that supports the current version of the spec :thinking:

**Comment by ai-oleynikov** [comment]
This package seems to support JSON Schema dratf 2020-12: https://www.nuget.org/packages/JsonSchema.Net

**Pull request closed without merging**